src/api/app/controllers/webui/staging/workflows_controller.rb

```diff
@@ -1,7 +1,9 @@
 class Webui::Staging::WorkflowsController < Webui::WebuiController
   VALID_STATES_WITH_REQUESTS = [:acceptable, :accepting, :review, :testing, :building, :failed, :unacceptable].freeze
 
-  before_action :require_login, except: [:show]
+  before_action :check_authorizable, except: [:show]
```

Comment:
In order to `authorize` we need a [user, record] pair, this is a problem when we need to deal with a not found record: if we return a `404` whilst fetching the record ( before `authorize` ) we leak the existence of the records to anonymous users (which I don't think it's the desired behaviour).

This implementation does 2 checks:
1. checks for the user presence by relying in the `initializer` `ensure_logged_in` check
2. runs the `authorize` as usual afterwards

This approach Is also a very easy plug in replacement of `require_login` with the Policy.

*If leaking resource existence to anonymous users is not an issue then we can keep the single authorize call and the `raise ActiveRecord::RecordNotFound` logic.*

Reply:
> _If leaking resource existence to anonymous users is not an issue then we can keep the single authorize call and the `raise ActiveRecord::RecordNotFound` logic._

It's not an issue. Let's keep the single `authorize` call and the `raise ActiveRecord::RecordNotFound` logic.